### ISSUE TITLE
ZipFile is not released after unzip with OneZip

### ISSUE THREAD

**Issue by florianperrot44** [coderaiser/node-onezip#4]
Here is my code :

```js
var extract = onezip.extract(path.join(downloadFolder,entry.filename), path.join(downloadFolder,fileSansExt(entry.filename)));


					extract.on('progress', (percent) => {
						entry.win.webContents.send('unzip-progress', entry, percent);
					});

					extract.on('error', (error) => {
						entry.progress = "ERROR";
						log.error("unzip-ko "+error)
						entry.win.webContents.send('unzip-ko', entry, i18n.t("errors.unzipFailed"));
					});

					extract.on('end', () => {
						// extraction is complete. make sure to handle the err
						log.info('UNZIP end');
						entry.progress = "UNZIPED";
						entry.win.webContents.send('unzip-ok', entry);
						// DELETE THE ZIP FILE
						fs.unlinkSync(path.join(downloadFolder,entry.filename));
						controleFiles(entry);
					});
```

The unlink do nothing because, OneZip doesn't release the ZipFile.
Maybe you should set AutoClose in yauzl call.
Thanks.

**Comment by coderaiser** [comment]
What error do you have? Could you please simplify example and make it runnable :)?

**Comment by florianperrot44** [comment]
I want to delete the zip file after uncompress it.
The problem is the zip file isn't deleted after the unzip.

Here is a more intuitive code :

`
var fs = require('fs')

function test() {
  var extract = onezip.extract("myzip.zip", "folder");

  extract.on('end', () => {
	console.info('UNZIP end');
	// DELETE THE ZIP FILE
	fs.unlinkSync(path.join(downloadFolder,entry.filename));
	try {
          if (fs.existsSync("myzip.zip")) {
               //file exists
               console.info("Problem : the file is'nt deleted !!")
          }
       } catch(err) {
         console.error(err)
       }
});
}



`

**Comment by coderaiser** [comment]
I can't reproduce this with:

```js

const fs = require('fs');
const onezip = require('onezip');

const extract = onezip.extract('./1.zip', '.');

extract.on('end', () => {
    console.info('UNZIP end');
    // DELETE THE ZIP FILE
    fs.unlinkSync('./1.zip');
    try {
        if (fs.existsSync('./1.zip')) {
            //file exists
            console.info('Problem : the file is\'nt deleted !!');
        }
    } catch(err) {}
});
```

I see only:

```sh
$ node 1
UNZIP end
```

Doesn't matter one file in archive or a couple.

**Comment by florianperrot44** [comment]
Ok, maybe it depends on the system (specially for fs library).
The programm is executed under Windows 10 and inside Electron (latest).
Please, check it with this configuration.

Moreover, i resolved the problem with a fork of OneZip.
I've changed this line (232) in onezip.js :  
   const autoClose = true;

Can you change that for a future release ?

Thanks!

**Comment by coderaiser** [comment]
Unfortunately I can’t check this, but I think it worth fixing. Could you please send PR for this?

**Comment by coderaiser** [comment]
Fixed with #5, landed in landed in [v4.2.1](https://github.com/coderaiser/node-onezip/releases/tag/v4.2.1) 🎉.
Is it works for you :)?
